### PR TITLE
Feature/refactor probe design

### PR DESCRIPTION
**Pull request by wesleymonte**

### CODE REVIEW

File: probes/src/main/java/cloud/fogbow/probes/core/utils/AppUtil.java [hunk 1]
```diff
@@ -54,4 +66,25 @@ public static Float percent(Integer dividend, Integer divisor) {
...
+    public static Properties readProperties() {
+        Properties properties = new Properties();
+        String confFilePath = System.getProperty(PropertiesHolder.CONF_FILE_PROPERTY);
+
+        try {
+            if (Objects.isNull(confFilePath)) {
+                confFilePath = "private/" + Constants.CONF_FILE;
+                properties.load(FogbowProbesApplication.class.getClassLoader()
+                    .getResourceAsStream(confFilePath));
+//                LOGGER.info("Configuration file found in default path " + confFilePath + ".");
```
Comment (joivo) [inline probes/src/main/java/cloud/fogbow/probes/core/utils/AppUtil.java]: Avoid these comments.

File: probes/src/main/java/cloud/fogbow/probes/core/probes/Probe.java [hunk 2]
```diff
@@ -0,0 +1,5 @@
+package cloud.fogbow.probes.core.probes;
+
+public interface Probe extends Runnable{
```
Comment (joivo) [inline probes/src/main/java/cloud/fogbow/probes/core/probes/Probe.java]: Run formatter here.

File: probes/src/main/java/cloud/fogbow/probes/core/controllers/scheduler/ProbeScheduler.java [hunk 3]
```diff
@@ -0,0 +1,8 @@
+package cloud.fogbow.probes.core.controllers.scheduler;
+
+import cloud.fogbow.probes.core.probes.Probe;
+import java.util.List;
+
+public interface ProbeScheduler {
```
Comment (joivo) [inline probes/src/main/java/cloud/fogbow/probes/core/controllers/scheduler/ProbeScheduler.java:10]: Document this.

File: probes/src/main/java/cloud/fogbow/probes/core/controllers/scheduler/ProbeScheduler.java [hunk 4]
```diff
@@ -0,0 +1,8 @@
...
+    void submitProbes(List<Probe> probes);
```
Comment (joivo) [inline probes/src/main/java/cloud/fogbow/probes/core/controllers/scheduler/ProbeScheduler.java:16]: Attempt to this method name. Doc this.

File: probes/src/main/java/cloud/fogbow/probes/core/probes/MetricCollector.java [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+import java.util.List;
+import java.util.Map;
+
+/**
+ * It is an entity in charge of making observations at every moment of time ). All observations are
+ * sent to the Fogbow Telemetry Aggregator by address {@link #ftaAddress} using {@link FtaSender}.
+ */
+public interface MetricCollector {
+
+    List<Metric> collect(Timestamp timestamp);
```
Comment (joivo) [inline probes/src/main/java/cloud/fogbow/probes/core/probes/MetricCollector.java:19]: Doc these methods.

File: probes/src/main/java/cloud/fogbow/probes/core/controllers/DefaultProbesController.java [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+
+    private ProbeScheduler probeScheduler;
+    private ProbeCreator probeCreator;
+
+    public DefaultProbesController() {
+        this.probeScheduler = new DefaultProbeScheduler();
+        this.probeCreator = new DockerProbeCreator();
+    }
+
+    public void submitProbes() {
```
Comment (joivo) [inline probes/src/main/java/cloud/fogbow/probes/core/controllers/DefaultProbesController.java:20]: Where are you submitting the probes?

**Review by joivo — CHANGES_REQUESTED** [review]

File: probes/src/main/java/cloud/fogbow/probes/core/utils/AppUtil.java (hunk 1 above)
Comment (wesleymonte) [reply probes/src/main/java/cloud/fogbow/probes/core/utils/AppUtil.java]: Done.

File: probes/src/main/java/cloud/fogbow/probes/core/probes/Probe.java (hunk 2 above)
Comment (wesleymonte) [reply probes/src/main/java/cloud/fogbow/probes/core/probes/Probe.java]: Done.

File: probes/src/main/java/cloud/fogbow/probes/core/controllers/DefaultProbesController.java (hunk 6 above)
Comment (wesleymonte) [reply probes/src/main/java/cloud/fogbow/probes/core/controllers/DefaultProbesController.java:20]: I am sending to the Probe Scheduler that the controller has.

File: probes/src/main/java/cloud/fogbow/probes/core/probes/MetricCollector.java (hunk 5 above)
Comment (wesleymonte) [reply probes/src/main/java/cloud/fogbow/probes/core/probes/MetricCollector.java:19]: Done.

File: probes/src/main/java/cloud/fogbow/probes/core/controllers/scheduler/ProbeScheduler.java (hunk 4 above)
Comment (wesleymonte) [reply probes/src/main/java/cloud/fogbow/probes/core/controllers/scheduler/ProbeScheduler.java:16]: Done.

File: probes/src/main/java/cloud/fogbow/probes/core/controllers/scheduler/ProbeScheduler.java (hunk 3 above)
Comment (wesleymonte) [reply probes/src/main/java/cloud/fogbow/probes/core/controllers/scheduler/ProbeScheduler.java:10]: Done.

**Review by joivo — APPROVED** [review]

**Pull request merged**